components/ConnectButton.js

```diff
@@ -0,0 +1,35 @@
+import { Button } from './ui/Buttons'
+import useWeb3 from './hooks/useWeb3'
+import { Body1 } from './ui/Typography'
+import { NetworkId } from '../util/constants'
+
+const ConnectButton = () => {
+  const { connect, accounts, error, provider, networkId } = useWeb3()
+
+  if (error) {
+    if (networkId !== NetworkId) {
+      return (
+        <Button
+          onClick={async () => {
+            await provider.request({
+              method: 'wallet_switchEthereumChain',
+              params: [{ chainId: NetworkId }],
+            })
+          }}
+        >
+          {error}
+        </Button>
+      )
+    }
+
+    return <Body1>{error}</Body1>
+  }
+
+  if (accounts.length) {
+    return <Body1>{accounts[0]}</Body1>
```

Comment:
are we ok with using the first account and not exposing the others?

Reply:
Usually there will only be one - unless the user connects multiple accounts in their metamask. I haven't seen many dApps that support multiple connected accounts at once though, usually they just take the first one, which is the one that's actively selected in metamask (or whatever wallet they are using).

Reply:
Sounds good!